### PR TITLE
Fix for Issue #11 - support for OSX x64

### PR DESCRIPTION
**Pull request by jhaynie**
This is a pull request for https://github.com/lloyd/node-memwatch/issues/11

This seems to work fine for me and my local machine by moving to node-gyp for building for node >=0.8.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,17 @@
 TESTS = test/*.js
 
-all: build
+all: test
```
Comment (lloyd) [inline Makefile]: don't understand what you were going for here ?

Comment (jhaynie) [inline Makefile]: oops this should have been `build test`.

**Comment by lloyd** [comment]
awesome!  This looks and works great on my linux box, I'll test on a couple more environments, merge, and push an updated v. of memwatch.

thank you @jhaynie, and hello again!

**Comment by lloyd** [comment]
hmmm.  so this actually doesn't work on osx 0.8 - because we need to, in the `binding.gyp` file, define 'NEW_COMPACTION_BEHAVIOR' for gcc when we're building on node 0.8.

master is now broken for 0.8 and I cannot figure out how to get gyp to dance this dance, tho I expect it'd be a pretty common requirement.

**Pull request merged**